stage #2.5/module #3 Spring Core/#5.2. Classpath Scanning and Managed Components/content.md

```diff
@@ -0,0 +1,238 @@
+
+```java
+
+@Service
+public class MusicRatingService {
+
+    private MusicRatesFinder musicRatesFinder;
+
+    public MusicRatingService(MusicRatesFinder musicRatesFinder) {
+        this.musicRatesFinder = musicRatesFinder;
+    }
+
+    // ...
+}
+
+@Repository
+public class MusicRatesFinderImpl implements MusicRatesFinder {
+    // ...
+}
+```
+
+To autodetect these classes and register the corresponding beans, you need to add `@ComponentScan` to
+your `@Configuration` class, where the `basePackages` attribute is a common parent package for the two classes.
+(Alternatively, you can specify a comma- or semicolon- or space-separated list that includes the parent package of each
+class.)
+
+```java
+
+@Configuration
+@ComponentScan(basePackages = "com.somepackage")
+public class Config {
+    // ...
+}
+```
+
+The following alternative uses XML:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans
+        https://www.springframework.org/schema/beans/spring-beans.xsd
+        http://www.springframework.org/schema/context
+        https://www.springframework.org/schema/context/spring-context.xsd">
+
+    <context:component-scan base-package="com.somepackage"/>
+
+</beans>
+```
+
+> The use of `<context:component-scan>` implicitly enables the functionality of `<context:annotation-config>`. There is
+> usually no need to include the `<context:annotation-config>` element when using `<context:component-scan>`.
+
+## Using Filters to Customize Scanning
+
+By default, classes annotated with `@Component`, `@Repository`, `@Service`, `@Controller`, `@Configuration`, or a custom
+annotation that itself is annotated with `@Component` are the only detected candidate components. However, you can
+modify and extend this behavior by applying custom filters. Add them as `includeFilters` or `excludeFilters` attributes
+of the `@ComponentScan` annotation. Each filter element requires the `type` and `expression` attributes.
+
+The following example shows the configuration ignoring all `@Repository` annotations and using "stub" repositories
+instead ([see more details about filtering options][2]):
+
+```java
+
+@Configuration
+@ComponentScan(basePackages = "com.somepackage",
+        includeFilters = @Filter(type = FilterType.REGEX, pattern = ".*Stub.*Repository"),
+        excludeFilters = @Filter(Repository.class))
+public class Config {
+    // ...
+}
+```
+
+## Defining Bean Metadata within Components
+
+Spring components can also contribute bean definition metadata to the container. You can do this with the same `@Bean`
+annotation used to define bean metadata within `@Configuration` annotated classes. The following example shows how to do
+so:
+
+```java
+
+@Component
+public class MyComponent {
```

Comment:
Logically and based on text above there should be '@Configuration' instead of '@Component'

Reply:
Will remove this section, thanks.